### PR TITLE
Use mass flowrate in WaterTAP Costing Package

### PR DESCRIPTION
**Pull request by bknueven**
## Fixes/Addresses:
Part of #402

## Changes proposed in this PR:
- Use mass flowrate in chemical costs
- Adjust full treatment train examples to reflect this
- Remove some hardcoded molecular weights

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by andrewlee94 — APPROVED** [review]
I don't see anything out of place here, but I am not that familiar with these models either.

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py [hunk 1]
```diff
@@ -44,7 +44,7 @@
 def test_ideal_naocl_mixer():
     model = run_ideal_naocl_mixer_example()
     assert model.fs.ideal_naocl_mixer_unit.dosing_rate.value == pytest.approx(
-        0.4, rel=1e-3
+        0.4e-6, rel=1e-3
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:47]: Were the units changed here, explaining the change in order of magnitude?

File: watertap/costing/watertap_costing_package.py [hunk 2]
```diff
@@ -193,16 +193,8 @@ def build_global_params(self):
...
-            * (pyo.units.kg / pyo.units.mol)
-            * (self.naocl_cost / self.naocl_purity)
-        )
-        self.defined_flows["CaOH2"] = (
-            74.093e-3
-            * (pyo.units.kg / pyo.units.mol)
-            * (self.caoh2_cost / self.caoh2_purity)
-        )
+        self.defined_flows["NaOCl"] = self.naocl_cost / self.naocl_purity
+        self.defined_flows["CaOH2"] = self.caoh2_cost / self.caoh2_purity
```
Comment (adam-a-a) [inline watertap/costing/watertap_costing_package.py:197]: Why isn't the molecular weight needed anymore?

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py (hunk 1 above)
Comment (adam-a-a) [reply watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:47]: (looks like ppm representation converted to mass fraction)

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/posttreatment_ideal_naocl_chlorination_block.py [hunk 3]
```diff
@@ -512,20 +512,26 @@ def build_ideal_naocl_mixer_unit(model):
...
+        return blk.dosing_rate == (
+            blk.naocl_stream.flow_mol[0]
             * blk.naocl_stream.mole_frac_comp[0, "OCl_-"]
-            * 74.44
-            * 1000
+            * blk.naocl_stream_state[0].params.get_component("OCl_-").mw
+        ) + (
+            blk.naocl_stream.flow_mol[0]
+            * blk.naocl_stream.mole_frac_comp[0, "Na_+"]
+            * blk.naocl_stream_state[0].params.get_component("Na_+").mw
```
Comment (bknueven) [inline watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/posttreatment_ideal_naocl_chlorination_block.py:534]: @adam-a-a this is where the units changed. Given the pretreatment example I think this was always meant to be in `kg/s`, but I'm not certain.

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py (hunk 1 above)
Comment (bknueven) [reply watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py:47]: Yes, here: https://github.com/watertap-org/watertap/pull/496/files#r855550900

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/posttreatment_ideal_naocl_chlorination_block.py (hunk 3 above)
Comment (adam-a-a) [reply watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/posttreatment_ideal_naocl_chlorination_block.py:534]: Got it.

File: watertap/costing/watertap_costing_package.py (hunk 2 above)
Comment (bknueven) [reply watertap/costing/watertap_costing_package.py:197]: Per @andrewlee94's suggestion, the molecular weight is part of the thermo configuration, so we use the conversion there in place of hardcoding it here.

Then the `flow` uses here should be in `kg/s` as opposed to `mol/s`, meaning the costing package doesn't also need to know the molecular weight.

Comment (adam-a-a) [reply watertap/costing/watertap_costing_package.py:197]: I think it makes sense.

**Review by adam-a-a — APPROVED** [review]
LGTM

**Pull request merged**